bluetooth_mesh/messages/silvair/rrule_scheduler.py

```diff
@@ -0,0 +1,241 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+#
+from enum import IntEnum
+from bluetooth_mesh.messages.util import EnumAdapter
+from bluetooth_mesh.messages.util import EnumSwitch as Switch
+from bluetooth_mesh.messages.util import Opcode, SwitchStruct
+
+from construct import (
+    BitsInteger,
+    ByteSwapped,
+    BitStruct,
+    GreedyRange,
+    Int8sl,
+    Int8ul,
+    Int16sl,
+    Int16ul,
+    Int32ul,
+    PrefixedArray,
+    Struct,
+    Optional,
+    this
+)
+
+
+class RRuleSchedulerOpcode(IntEnum):
+    SILVAIR_RRULE_SCHEDULER = 0xE83601
+
+
+class RRuleSchedulerSubOpcode(IntEnum):
+    RULES_LIST_GET = 0x00
+    RULES_LIST_STATUS = 0x01
+    REGISTER_MAX_SIZE_GET = 0x02
+    REGISTER_MAX_SIZE_STATUS = 0x03
+    SCHEDULER_MODE_GET = 0x04
+    SCHEDULER_MODE_SET = 0x05
+    SCHEDULER_MODE_STATUS = 0x06
+    SCHEDULE_REGISTER_ENTRY_GET = 0x07
+    SCHEDULE_REGISTER_ENTRY_SET = 0x08
+    SCHEDULE_REGISTER_ENTRY_DELETE = 0x09
+    SCHEDULE_REGISTER_ENTRY_DELETE_ALL = 0x0A
+    SCHEDULE_REGISTER_ENTRY_STATUS = 0x0B
+    SCHEDULE_REGISTER_LIST_GET = 0x0C
+    SCHEDULE_REGISTER_LIST_STATUS = 0x0D
+
+
+class SchedulerMode(IntEnum):
+    DISABLED = 0x00
+    ENABLED = 0x01
+
+
+class Day(IntEnum):
+    PREVIOUS_SUNDAY = 0xEF
+    PREVIOUS_SATURDAY = 0xFA
+    PREVIOUS_FRIDAY = 0xFB
+    PREVIOUS_THURSDAY = 0xFC
+    PREVIOUS_WEDNESDAY = 0xFD
+    PREVIOUS_TUESDAY = 0xFE
+    PREVIOUS_MONDAY = 0xFF
+    NEXT_MONDAY = 0x01
+    NEXT_TUESDAY = 0x02
+    NEXT_WEDNESDAY = 0x03
+    NEXT_THURSDAY = 0x04
+    NEXT_FRIDAY = 0x05
+    NEXT_SATURDAY = 0x06
+    NEXT_SUNDAY = 0x07
+
+
+class StatusCode(IntEnum):
+    SUCCESS = 0x00
+    INVALID_SLOT = 0x01
+    INVALID_SLOT_PARAMETERS = 0x02
+    INVALID_RULE = 0x03
+    INVALID_RULE_VALUE = 0x04
+    REGISTER_SIZE_EXCEEDED = 0x05
+    STORAGE_FAILURE = 0x06
+
+
+class RuleID(IntEnum):
+    FREQ = 0x00
+    UNTIL = 0x01
+    COUNT = 0x02
+    INTERVAL = 0x03
+    BYSECOND = 0x04
+    BYMINUTE = 0x05
+    BYHOUR = 0x06
+    BYDAY = 0x07
+    BYMONTHDAY = 0x08
+    BYYEARDAY = 0x09
+    BYWEEKNO = 0x0A
+    BYMONTH = 0x0B
+    BYSETPOS = 0x0C
+    DTSTART = 0x0D
+    EXPLICIT_RDATE = 0x0E
+    EXCLUSIONS_RDATE = 0x0F
+
+
+class Freq(IntEnum):
+    SECONDLY = 0x00
+    MINUTELY = 0x01
+    HOURLY = 0x02
+    DAILY = 0x03
+    WEEKLY = 0x04
+    MONTHLY = 0x05
+    YEARLY = 0x06
+
+
+Slots = Struct(
+    "slot_id" / Int16ul,
+    "element" / Int8ul,
+    "slot_parameter" / PrefixedArray(Int8ul, Int8ul)
```

Comment:
I think this should be just `Prefixed`, not `PrefixedArray`, as the field is an opaque buffer and should parse as `bytes`, not a `List[int]`